### ISSUE TITLE
Huge Form very slow when "Change"

### ISSUE THREAD

**Issue by silentbalanceyh** [erikras/redux-form#2181]
I'm not sure if it's an issue, but I got a problem when i put about 30 fields in the form. Every time when I type in Text Box, I found the "CHANGE" event triggered and there are some delay. When I only passed onBlur and onFocus in input as following segment:
`onBlur={input.onBlur}
onFocus={input.onFocus}`
The onChange problem has been resolved, but I found reset button is not working. 
How to resolve this kind of issue ?

**Comment by JakeLin1024** [comment]
Not sure if you are mapping any values from state to props which might cause re-rendering every time you change your fields...

**Comment by idoo** [comment]
@JakeLin1024 do you know some solutions how to bind it with `debounce`?

**Comment by gustavohenke** [comment]
Hey guys, is anyone able to provide a minimal, reproducible example, so we can give a look?
I know about #2308.

**Comment by silentbalanceyh** [comment]
Now I removed onChange event from input label except some autocompleted features. Although this action could not improve performance more but faster, when I click the field and trigger some remove api calling, onFocus/onBlur function also took some time but it could be accepted. From my experience, here are some comments:
1. When I selected the field and auto filling linked field, the AutoFill event is triggered many times by our field number. If I linked around 10 fields, this event has been triggered 10 times. I'm trying to find a way to do mass filling
2. I have corrected all the re-render events that is wasted, but compared with Pure React-Redux controls such as user-defined menu/bar etc, redux-form is still slower than that.
